### PR TITLE
Fix hardcoded path to link.exe

### PR DESCRIPTION
**Pull request by UK992**
Fixes https://github.com/servo/servo/issues/15228

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15319)

### CODE REVIEW

File: components/script/build.rs [hunk 1]
```diff
@@ -37,7 +37,12 @@ fn main() {
         // for reasons that I don't understand.  If we just give
         // link.exe, it tries to use script-*/out/link.exe, which of
         // course does not exist.
-        build.define("CMAKE_LINKER", "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\link.exe");
+        let platform = env::var("PLATFORM").expect("Environment variable 'PLATFORM' is not defined");
+        let link_dir = PathBuf::from(env::var("VCINSTALLDIR").unwrap())
```
Comment (nox) [inline components/script/build.rs]: This is the executable path but you named it `_dir`.

Comment (UK992) [inline components/script/build.rs]: fixed

File: components/script/build.rs [hunk 2]
```diff
@@ -37,7 +37,12 @@ fn main() {
         // for reasons that I don't understand.  If we just give
         // link.exe, it tries to use script-*/out/link.exe, which of
         // course does not exist.
-        build.define("CMAKE_LINKER", "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\link.exe");
+        let platform = env::var("PLATFORM").expect("Environment variable 'PLATFORM' is not defined");
+        let link_path = PathBuf::from(env::var("VCINSTALLDIR").unwrap())
+                                .join("bin")
+                                .join(if platform.to_uppercase() == "X64" { "amd64" } else { "" })
```
Comment (emilio) [inline components/script/build.rs]: nit: You could use `AsciiExt` and do `if platform.eq_ignore_ascii_case("x64")`, but not that it matters a lot.

**Review by emilio — COMMENTED** [review]
This looks ok to me, but I have no way to verify the paths are correct, r? @larsbergstrom maybe?

**Comment by larsbergstrom** [comment]
If it passes AppVeyor, squash and r=me. Thanks!

**Comment by KiChjang** [comment]
Still requires squashing.

**Comment by emilio** [comment]
Thanks @UK992! :)

**Pull request merged**